src/audio/module_adapter/library/native_system_agent.c

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Pawel Dobrowolski <pawelx.dobrowolski@intel.com>
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <utilities/array.h>
+#include <native_system_agent.h>
+
+/* The create_instance_f is a function call type known in module. The module entry_point
+ * points to this type of function which starts module creation.
+ */
+
+void SystemServiceLogMessage(AdspLogPriority log_priority, uint32_t log_entry,
+			     AdspLogHandle const *log_handle, uint32_t param1,
+			     uint32_t param2, uint32_t param3, uint32_t param4);
+
+AdspErrorCode SystemServiceSafeMemcpy(void *RESTRICT dst, size_t maxlen,
```

Comment:
I suspect these are IADK APIs to be used in the IADK module layer ? if so, It's probably worth spelling this out in the header so that everyone knows the APIs have external users for compatibility.